src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java

```diff
@@ -121,38 +128,46 @@ public PackageObjectFactory(String packageName, ClassLoader moduleClassLoader) {
      */
     @Override
     public Object createModule(String name) throws CheckstyleException {
-        Object instance = createObjectFromMap(name);
+        final String nameCheck = name + CHECK_SUFFIX;
+        Object instance = createObject(name);
         if (instance == null) {
-            instance = createObjectWithIgnoringProblems(name, getAllPossibleNames(name));
+            instance = createObject(nameCheck);
         }
         if (instance == null) {
-            final String nameCheck = name + CHECK_SUFFIX;
-            instance = createObjectWithIgnoringProblems(nameCheck, getAllPossibleNames(nameCheck));
-            if (instance == null) {
-
-                final String attemptedNames = joinPackageNamesWithClassName(name, packages)
-                        + STRING_SEPARATOR + nameCheck + STRING_SEPARATOR
-                        + joinPackageNamesWithClassName(nameCheck, packages);
-                final LocalizedMessage exceptionMessage = new LocalizedMessage(0,
-                    Definitions.CHECKSTYLE_BUNDLE, UNABLE_TO_INSTANTIATE_EXCEPTION_MESSAGE,
-                    new String[] {name, attemptedNames}, null, getClass(), null);
-                throw new CheckstyleException(exceptionMessage.getMessage());
+            instance = createObjectFromMap(name, NAME_TO_FULL_MODULE_NAME);
+        }
+        if (instance == null) {
+            if (thirdPartyNameToFullModuleName == null) {
+                thirdPartyNameToFullModuleName =
+                        generateThirdPartyNameToFullModuleName(moduleClassLoader);
             }
+            instance = createObjectFromMap(name, thirdPartyNameToFullModuleName);
+        }
+        if (instance == null) {
+            final String attemptedNames = joinPackageNamesWithClassName(name, packages)
+                    + STRING_SEPARATOR + nameCheck + STRING_SEPARATOR
+                    + joinPackageNamesWithClassName(nameCheck, packages);
+            final LocalizedMessage exceptionMessage = new LocalizedMessage(0,
+                Definitions.CHECKSTYLE_BUNDLE, UNABLE_TO_INSTANTIATE_EXCEPTION_MESSAGE,
+                new String[] {name, attemptedNames}, null, getClass(), null);
+            throw new CheckstyleException(exceptionMessage.getMessage());
         }
         return instance;
     }
 
     /**
-     * Create object with the help of Checkstyle NAME_TO_FULL_MODULE_NAME map.
+     * Create object with the help of Checkstyle NAME_TO_FULL_MODULE_NAME map or third party map.
```

Comment:
We should describe method as what it does, not what it does where it is called, otherwise we will have to update this for every new call that is added.
imo: `Checkstyle NAME_TO_FULL_MODULE_NAME map or third party map` -> `the supplied map.`